mypy/main.py

```diff
@@ -102,120 +104,152 @@ def process_options(args: List[str]) -> Tuple[List[BuildSource], Options]:
+    options.custom_typing_module = args.custom_typing
 
-    if help:
-        usage()
+    # Set build flags.
+    if args.py2:
+        options.pyversion = defaults.PYTHON2_VERSION
 
-    if ver:
-        version()
+    if args.python_version is not None:
+        options.pyversion = args.python_version
 
-    if not args:
-        usage('Missing target file or module')
+    if args.verbose:
+        options.build_flags.append(build.VERBOSE)
 
-    if options.python_path and options.pyversion[0] == 2:
-        usage('Python version 2 (or --py2) specified, '
-              'but --use-python-path will search in sys.path of Python 3')
+    if args.stats:
+        options.build_flags.append(build.DUMP_TYPE_STATS)
 
-    if build.FAST_PARSER in options.build_flags and options.pyversion[0] == 2:
-        usage('The experimental fast parser is only compatible with Python 3, '
-              'but Python 2 specified.')
+    if args.inferstats:
+        options.build_flags.append(build.DUMP_INFER_STATS)
 
-    targets = []
-    for arg in args:
-        if arg.endswith(PY_EXTENSIONS):
-            targets.append(BuildSource(arg, crawl_up(arg)[1], None))
-        elif os.path.isdir(arg):
-            targets.extend(expand_dir(arg))
-        else:
-            targets.append(BuildSource(arg, None, None))
-    return targets, options
+    if args.silent_imports:
+        options.build_flags.append(build.SILENT_IMPORTS)
+
+    if args.disallow_untyped_calls:
+        options.build_flags.append(build.DISALLOW_UNTYPED_CALLS)
+
+    if args.disallow_untyped_defs:
+        options.build_flags.append(build.DISALLOW_UNTYPED_DEFS)
+
+    # experimental
+    if args.fast_parse:
+        options.build_flags.append(build.FAST_PARSER)
+
+    # Set reports.
+    for flag, val in vars(args).items():
+        if flag.endswith('_report') and val is not None:
+            report_type = flag[:-7].replace('_', '-')
+            report_dir = val
+            options.report_dirs[report_type] = report_dir
+
+    # Set target.
+    if args.module:
+        options.build_flags.append(build.MODULE)
+        return [BuildSource(None, args.module, None)], options
+    elif args.package:
+        options.build_flags.append(build.MODULE)
+        lib_path = [os.getcwd()] + build.mypy_path()
+        targets = build.find_modules_recursive(args.package, lib_path)
+        if not targets:
+            fail("Can't find package '{}'".format(args.package))
+        return targets, options
+    elif args.command:
+        options.build_flags.append(build.PROGRAM_TEXT)
+        return [BuildSource(None, None, args.command)], options
+    else:
+        targets = []
+        for f in args.files:
+            if f.endswith(PY_EXTENSIONS):
+                targets.append(BuildSource(f, crawl_up(f)[1], None))
+            elif os.path.isdir(f):
+                targets.extend(expand_dir(f))
+            else:
+                targets.append(BuildSource(f, None, None))
```

Comment:
Would be nice to check there's at most one of these (I guess this is for scripts), since they all get the module name `__main__` and you can't have duplicate module names. There's a check for that in build.py but it's a bit ugly so late in the game (it used to be an assert :-).